### ISSUE TITLE
version 1.2.3 still disconnects after some time (12h?)

### ISSUE THREAD

**Issue by mazi1987** [JohNan/homeassistant-wellbeing#126]
It seems that the newest version token refresh doesn't not work, I need to re-add the device every day to HA

**Comment by piolec66** [comment]
In my case, device is every day unavailable between  ~15:00 and ~02:00. Every day... Also refresh token?

**Comment by modo1302** [comment]
in the Electrolux dev page, you can see this info:
![image](https://github.com/user-attachments/assets/c4fbab47-76c5-4747-a4e6-67c87733cbde)

thats 12h.
that sucks!

**Comment by HonkStonk** [comment]
Hmm - not sure how the token refresh/expiration is supposed to work but now that I found this issue it seems I'm experiencing exactly the same except my history shows its unavailable more or less 7h out of 24h every day.

What I also notice in this repeating pattern is that after 7h of unavailability - it gets roughly 12+4h of availability divided in two "blocks" of "Idle"... so for sure there seems to be a 12h timer somewhere but I don't get what gives only 4h of availability after the 12...

![periodicallyUnavailable](https://github.com/user-attachments/assets/27101bb0-0dad-4ab2-b613-df4688cd06ea)


Suggestions?

**Comment by PsychoV3** [comment]
Defo experiencing the same issue here, it appears the refresh token timer is off, as in the evening - No access, but by the morning it is running again.

**Comment by HonkStonk** [comment]
Yeah… but how to fix? Would it help to create new tokens and set some
settings upon creation that didn’t exist before?
